### DISCUSSION TITLE
ERR_NGROK_8012

### DISCUSSION THREAD

**Discussion by bido75** [ngrok/ngrok#27] [Bug Reports]
![ngrok](https://github.com/user-attachments/assets/d4c0f841-2da0-438e-8fc1-3c54e822b2cf)

Traffic was successfully tunneled to the ngrok agent, but the agent failed to establish a connection to the upstream web service at [http://localhost:80](http://localhost/). The error encountered was:

dial tcp [::1]:80: connectex: No connection could be made because the target machine actively refused it.

[Get help with this error](https://ngrok.com/docs/errors/err_ngrok_8012)
If you're the developer of this page

On the machine where the ngrok agent is running, make sure a web service is running on [http://localhost:80](http://localhost/). Try to cURL or open the address in a browser to see that you get the correct response.

Check out the docs to get [help with this error](https://ngrok.com/docs/errors/err_ngrok_8012).
If you're a visitor of this page

Wait a few minutes and
. If that still doesn't work, please contact the developer of this page for more information.

**Comment by russorat** [comment]
@bido75 thanks for this. is there an actual question here?
